1436790/1438462: Part 38: 19 files with MSDN link fixes.

DennisLee-DennisLee opened the pull request:
Contributes to PR #[2037](https://github.com/dotnet/docs/issues/2037) and applies to the following work items:
- [User Story 1436790: Content Update: update MSDN links for .NET content](https://mseng.visualstudio.com/TechnicalContent/_workitems/edit/1436790)
- [Task 1438462: Update MSDN links for .NET content](https://mseng.visualstudio.com/TechnicalContent/_workitems/edit/1438462)

DennisLee-DennisLee committed:
Another 20 files with MSDN link fixes.

DennisLee-DennisLee committed:
Merging changes pulled from master.

mairaw commented on docs/framework/data/adonet/ado-net-technology-options-and-guidelines.md:
```diff
@@ -22,7 +22,7 @@ The ADO.NET Data Platform is a multi-release strategy to decrease the amount of
   
 ### EntityClient  
- EntityClient is a new .NET Framework data provider used for interacting with an Entity Data Model. EntityClient follows the .NET Framework data provider pattern of exposing <xref:System.Data.EntityClient.EntityConnection> and <xref:System.Data.EntityClient.EntityCommand> objects that return an <xref:System.Data.EntityClient.EntityDataReader>. EntityClient works with the Entity SQL language, prov ...
+ EntityClient is a new .NET Framework data provider used for interacting with an Entity Data Model. EntityClient follows the .NET Framework data provider pattern of exposing <xref:System.Data.EntityClient.EntityConnection> and <xref:System.Data.EntityClient.EntityCommand> objects that return an <xref:System.Data.EntityClient.EntityDataReader>. EntityClient works with the Entity SQL language, prov ...
```
Maybe replace link 49202ab9-ac98-4b4b-a05c-140e422bf527 with https://docs.microsoft.com/en-us/dotnet/framework/data/adonet/ef/entityclient-provider-for-the-entity-framework?

mairaw commented on docs/framework/data/adonet/sql/sql-server-express-security.md:
```diff
@@ -25,7 +25,7 @@ Microsoft SQL Server Express Edition (SQL Server Express) is based on Microsoft
 |||  
 |-|-|  
-|[Connecting to SQL Server Express](https://msdn.microsoft.com/library/ms165679.aspx) in SQL Server Books Online|Describes how to use SQL Server Express Edition on a network.|  
+|[Connecting to SQL Server Express](https://docs.microsoft.com/previous-versions/sql/sql-server-2005/ms165679(v=sql.90)) in SQL Server Books Online|Describes how to use SQL Server Express Edition on a network.|  
```
This version is so old. SQL Server express version now is 2017. Better to remove it.

```suggestion
```

mairaw commented on docs/framework/winforms/controls/how-to-set-default-cell-styles-for-the-windows-forms-datagridview-control.md:
```diff
@@ -13,7 +13,7 @@ ms.assetid: 1aaaca43-5340-447e-99c0-9177d9776aa1
 With the <xref:System.Windows.Forms.DataGridView> control, you can specify default cell styles for the entire control and for specific columns and rows. These defaults filter down from the control level to the column level, then to the row level, then to the cell level. If a particular <xref:System.Windows.Forms.DataGridViewCellStyle> property is not set at the cell level, the default property se ...
   
- There is extensive support for this task in Visual Studio.  Also see [How to: Set Default Cell Styles and Data Formats for the Windows Forms DataGridView Control Using the Designer](https://msdn.microsoft.com/library/95y5fz2x\(v=vs.110\)).  
+ There is extensive support for this task in Visual Studio.  Also see [How to: Set Default Cell Styles and Data Formats for the Windows Forms DataGridView Control Using the Designer](./default-cell-styles-datagridview.md).  
```
```suggestion
 There is extensive support for this task in Visual Studio.  Also see [How to: Set Default Cell Styles and Data Formats for the Windows Forms DataGridView Control Using the Designer](default-cell-styles-datagridview.md).  
```

mairaw commented on docs/framework/winforms/controls/how-to-set-font-and-color-styles-in-the-windows-forms-datagridview-control.md:
```diff
@@ -16,7 +16,7 @@ You can specify the visual appearance of cells within a <xref:System.Windows.For
  The following procedures demonstrate basic customization of cell appearance using the <xref:System.Windows.Forms.DataGridView.DefaultCellStyle%2A> property. Every cell in the control inherits the styles specified through this property unless they are overridden at the column, row, or cell level. For an example of style inheritance, see [How to: Set Default Cell Styles for the Windows Forms DataG ...
   
- There is extensive support for this task in Visual Studio.  Also see [How to: Set Default Cell Styles and Data Formats for the Windows Forms DataGridView Control Using the Designer](https://msdn.microsoft.com/library/95y5fz2x\(v=vs.110\)).  
+ There is extensive support for this task in Visual Studio.  Also see [How to: Set Default Cell Styles and Data Formats for the Windows Forms DataGridView Control Using the Designer](./default-cell-styles-datagridview.md).  
```
```suggestion
 There is extensive support for this task in Visual Studio.  Also see [How to: Set Default Cell Styles and Data Formats for the Windows Forms DataGridView Control Using the Designer](default-cell-styles-datagridview.md).  
```

mairaw commented on docs/framework/winforms/controls/hscrollbar-and-vscrollbar-controls-overview-windows-forms.md:
```diff
@@ -32,5 +32,5 @@ Windows Forms <xref:System.Windows.Forms.ScrollBar> controls are used to provide
 - <xref:System.Windows.Forms.HScrollBar>
 - <xref:System.Windows.Forms.VScrollBar>
-- [Additions to Windows Forms for the .NET Framework 2.0](https://msdn.microsoft.com/library/c61a923d-3d6a-4c8c-820c-e94c83f3f9a8)
+- [Additions to Windows Forms for the .NET Framework 2.0](https://docs.microsoft.com/previous-versions/visualstudio/visual-studio-2010/35f2fe4h(v=vs.100))
```
```suggestion
```

mairaw commented on docs/framework/winforms/controls/index.md:
```diff
@@ -33,5 +33,5 @@ As you design and modify the user interface of your Windows Forms applications,
  Provides an overview of developing Windows-based applications.  
   
- [Windows Forms Walkthroughs](https://msdn.microsoft.com/library/fd44d13d-4733-416f-aefc-32592e59e5d9)  
+ [Windows Forms Walkthroughs](https://docs.microsoft.com/previous-versions/visualstudio/visual-studio-2010/zftbwa2b(v=vs.100))  
```
```suggestion
```

mairaw commented on docs/framework/winforms/controls/index.md:
```diff
@@ -33,5 +33,5 @@ As you design and modify the user interface of your Windows Forms applications,
   
- [Windows Forms Walkthroughs](https://msdn.microsoft.com/library/fd44d13d-4733-416f-aefc-32592e59e5d9)  
+ [Windows Forms Walkthroughs](https://docs.microsoft.com/previous-versions/visualstudio/visual-studio-2010/zftbwa2b(v=vs.100))  
  Lists walkthrough topics about Windows Forms and controls.
```
```suggestion
```

mairaw commented on docs/framework/winforms/controls/margin-and-padding-in-windows-forms-controls.md:
```diff
@@ -19,7 +19,7 @@ Precise placement of controls on your form is a high priority for many applicati
  ![Padding And Margin for Windows Forms Controls](../../../../docs/framework/winforms/controls/media/vs-winformpadmargin.gif "VS_WinFormPadMargin")  
   
- There is design-time support for this feature in Visual Studio.  Also see [Walkthrough: Laying Out Windows Forms Controls with Padding, Margins, and the AutoSize Property](https://msdn.microsoft.com/library/3z3f9e8b\(v=vs.110\)).  
+ There is design-time support for this feature in Visual Studio.  Also see [Walkthrough: Laying Out Windows Forms Controls with Padding, Margins, and the AutoSize Property](./windows-forms-controls-padding-autosize.md).  
```
```suggestion
 There is design-time support for this feature in Visual Studio. Also see [Walkthrough: Laying Out Windows Forms Controls with Padding, Margins, and the AutoSize Property](windows-forms-controls-padding-autosize.md).  
```

mairaw commented on docs/framework/winforms/controls/prevent-row-addition-and-deletion-datagridview.md:
```diff
@@ -13,7 +13,7 @@ ms.assetid: ef9539ce-539b-404e-84b6-ac282b64b88c
 Sometimes you will want to prevent users from entering new rows of data or deleting existing rows in your <xref:System.Windows.Forms.DataGridView> control. The <xref:System.Windows.Forms.DataGridView.AllowUserToAddRows%2A> property indicates whether the row for new records is present at the bottom of the control, while the <xref:System.Windows.Forms.DataGridView.AllowUserToDeleteRows%2A> property ...
   
- There is support for this task in Visual Studio.  Also see [How to: Prevent Row Addition and Deletion in the Windows Forms DataGridView Control Using the Designer](https://msdn.microsoft.com/library/k5c88sw3\(v=vs.110\)).  
+ There is support for this task in Visual Studio.  Also see [How to: Prevent Row Addition and Deletion in the Windows Forms DataGridView Control Using the Designer](./prevent-row-addition-and-deletion-in-the-datagrid-using-the-designer.md).  
```
```suggestion
 There is support for this task in Visual Studio. Also see [How to: Prevent Row Addition and Deletion in the Windows Forms DataGridView Control Using the Designer](prevent-row-addition-and-deletion-in-the-datagrid-using-the-designer.md).  
```

mairaw commented on docs/framework/winforms/controls/run-procedures-at-set-intervals-with-wf-timer-component.md:
```diff
@@ -17,14 +17,14 @@ ms.assetid: 8025247a-2de4-4d86-b8ab-a8cb8aeab2ea
 ### To run a procedure at set intervals with the Timer component  
   
-1.  Add a <xref:System.Windows.Forms.Timer> to your form. See the following Example section for an illustration of how to do this programmatically. Visual Studio also has support for adding components to a form. Also see [How to: Add Controls Without a User Interface to Windows Forms](https://msdn.microsoft.com/library/becyw7bz\(v=vs.110\)).  
+1.  Add a <xref:System.Windows.Forms.Timer> to your form. See the following Example section for an illustration of how to do this programmatically. Visual Studio also has support for adding components to a form. Also see [How to: Add Controls Without a User Interface to Windows Forms](./how-to-add-controls-without-a-user-interface-to-windows-forms.md).  
```
```suggestion
1.  Add a <xref:System.Windows.Forms.Timer> to your form. See the following Example section for an illustration of how to do this programmatically. Visual Studio also has support for adding components to a form. Also see [How to: Add Controls Without a User Interface to Windows Forms](how-to-add-controls-without-a-user-interface-to-windows-forms.md).  
```

mairaw commented on docs/framework/winforms/controls/run-procedures-at-set-intervals-with-wf-timer-component.md:
```diff
@@ -17,14 +17,14 @@ ms.assetid: 8025247a-2de4-4d86-b8ab-a8cb8aeab2ea
 > [!NOTE]
 >  There are some limitations when using the <xref:System.Windows.Forms.Timer> component. For more information, see [Limitations of the Windows Forms Timer Component's Interval Property](../../../../docs/framework/winforms/controls/limitations-of-the-timer-component-interval-property.md).  
   
 ### To run a procedure at set intervals with the Timer component  
```
not related to your changes but fixing it since I noticed this small issue with the header level jumping from H1 to h3 

```suggestion
## To run a procedure at set intervals with the Timer component  
```

mairaw reviewed (changes requested):
Thanks @DennisLee-DennisLee. Left a few suggestions.

mairaw committed:
Apply suggestions from code review

Adding Maira's 10 autocommittable suggestions.

Co-Authored-By: DennisLee-DennisLee <v-dele@microsoft.com>

DennisLee-DennisLee committed:
Merge branch 'master' of https://github.com/dotnet/docs into v-dele-1438462-038

DennisLee-DennisLee committed:
Adding back an EntityClient link per Maira.

mairaw commented on docs/framework/data/adonet/data-providers.md:
```diff
@@ -149,7 +149,7 @@ using System.Data.OracleClient;
 |[!INCLUDE[dnprdnshort](../../../../includes/dnprdnshort-md.md)] Data Provider for Oracle|Recommended for middle and single-tier applications that use Oracle data sources.|  
   
 ## EntityClient Provider  
- The EntityClient provider is used for accessing data based on an Entity Data Model (EDM). Unlike the other .NET Framework data providers, it does not interact directly with a data source. Instead, it uses Entity SQL to communicate with the underlying data provider. For more information, see [EntityClient and Entity SQL](https://msdn.microsoft.com/library/49202ab9-ac98-4b4b-a05c-140e422bf527).  
```
same change would apply here too. I should have been more clear that the 49202ab9-ac98-4b4b-a05c-140e422bf527 link happened more than once in the PR.

DennisLee-DennisLee committed:
Update data-providers.md

Update **docs/framework/data/adonet/data-providers.md** with a current EntityClient link.

mairaw reviewed (approved):
Thanks @DennisLee-DennisLee!
